google_guest_agent/instance_setup.go

```diff
@@ -193,6 +241,307 @@ func agentInit(ctx context.Context) {
 	}
 }
 
+func setMultiQueue(totalCPUs int) error {
+	devices, err := filepath.Glob(virtioNetDevs)
+	if err != nil {
+		return err
+	}
+	for _, dev := range devices {
+		if err := enableMultiQueue(dev); err != nil {
+			logger.Warningf("Could not enable multi queue for %s.", dev)
+			return err
+		}
+		if err = setQueueNumForDevice(dev); err != nil {
+			logger.Warningf("Could not set queue num for %s.", dev)
+			return err
+		}
+	}
+	// Set smp_affinity properly for gvnic queues. '-ntfy-block.' is unique to gve and will not affect virtio queues.
+	if err = setSMPAffinityForGVNIC(totalCPUs); err != nil {
+		logger.Warningf("Could not set smp_affinity for gvnic.")
+		return err
+	}
+	return nil
+}
+
+func setSMPAffinityForGVNIC(totalCPUs int) error {
+	irqDirs, err := filepath.Glob(irqDirPath)
+	if err != nil {
+		return err
+	}
+	for _, irq := range irqDirs {
+		blocks, err := filepath.Glob(irq + "/*-ntfy-block.*")
+		if err != nil {
+			return err
+		}
+		if len(blocks) > 0 && isFile(irq+"/affinity_hint") {
+			if err = copyFile(irq+"/affinity_hint", irq+"/smp_affinity"); err != nil {
+				return err
+			}
+		}
+	}
+	XPS, err := filepath.Glob(xpsCPU)
```

Comment:
although the shell script was not broken into functions, the code from here to end of function is really unrelated to setting smp affinity on gvnic queues. it is better named something like configureXPS or configureTransmitPacketSteering